packaging/packager

```diff
@@ -66,14 +65,16 @@ function package_libc_via_pacman {
 
 function package_libc_via_dpkg() {
     if type dpkg-query > /dev/null 2>&1; then
-        if [[ $(dpkg-query --listfiles libc6 | wc -l) -gt 0 ]]; then
-            dpkg-query --listfiles libc6 | sed -E '/\.so$|\.so\.[0-9]+$/!d'
+       architecture=$(dpkg --print-architecture)
```

Comment:
I just [added](https://github.com/awslabs/aws-lambda-cpp/pull/131) Arch Linux support in the packager the other day, and I switched to using `uname -m` for checking the system architecture for rpm-based distros.

Reply:
Doesn't work on Debian though:

```
$ uname -m
x86_64
$ dpkg --print-architecture
amd64
```

Using `x86_64` as the architecture does not work with Debian.

Reply:
Oh, interesting.

And I just took a look with a fedora image from docker hub, and it turns out `arch` comes with `coreutils` on some distros:
```
$ arch --help
Usage: arch [OPTION]...
Print machine architecture.

      --help     display this help and exit
      --version  output version information and exit

GNU coreutils online help: <https://www.gnu.org/software/coreutils/>
Report any translation bugs to <https://translationproject.org/team/>
Full documentation <https://www.gnu.org/software/coreutils/arch>
or available locally via: info '(coreutils) arch invocation'
```
And the [docs](https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html#arch-invocation) say:
> `arch` prints the machine hardware name, and is equivalent to `uname -m`

Reply:
Just want to give you a heads up, I just opened PR #136, which has some simplifications in the same area as this PR.

Reply:
I cherry-picked your changes from here into #136: d63870c